### PR TITLE
Add support for server-side apply

### PR DESCRIPTION
**Pull request by tintoy**
Note - still using the Yaml serialization process described in the commit notes because JSON.NET's custom attributes are more flexible than YamlDotNet's ones (and YamlDotNet can't handle reuse of exiting collection instances in read-only properties, which better fits the KubeClient model philosophy).

### CODE REVIEW

**Comment by felixfbecker** [comment]
> Note - still using the Yaml serialization process described in the commit notes because JSON.NET's custom attributes are more flexible than YamlDotNet's ones (and YamlDotNet can't handle reuse of exiting collection instances in read-only properties, which better fits the KubeClient model philosophy).

Then why not always serialize to JSON?

**Comment by tintoy** [comment]
I guess because the JSON payloads are harder to read than regular YAML and my instincts tell me that the closer the client comes to actually supplying the request body in the format shown in the docs (I know that JSON is technically valid YAML, but how often does anyone actually test using it like that), the fewer the weird edge cases we may someday hit. But I’m open to having my mind changed :)

File: src/KubeClient/ResourceClients/DynamicResourceClient.cs [hunk 1]
```diff
@@ -276,6 +278,145 @@ public async Task<KubeResourceV1> Patch(string name, string kind, string apiVers
...
+        /// <param name="kubeNamespace">
+        ///     The (optional) name of a Kubernetes namespace containing the resources.
+        /// </param>
+        /// <param name="cancellationToken">
+        ///     An optional <see cref="CancellationToken"/> that can be used to cancel the request.
+        /// </param>
+        /// <returns>
+        ///     A <see cref="KubeResourceV1"/> representing the updated resource.
+        /// </returns>
+        public async Task<KubeResourceV1> ApplyYaml(string name, string kind, string apiVersion, string yaml, string fieldManager, string kubeNamespace = null, CancellationToken cancellationToken = default)
```
Comment (felixfbecker) [inline src/KubeClient/ResourceClients/DynamicResourceClient.cs]: One small improvement - for apply you often need to supply `force=true` as query parameter to take ownership of a field. Could we have a parameter for that?

Comment (felixfbecker) [reply src/KubeClient/ResourceClients/DynamicResourceClient.cs]: @tintoy this would be the only blocker to use server side apply and supporting `kubectl apply` in PSKubectl :)

Comment (tintoy) [reply src/KubeClient/ResourceClients/DynamicResourceClient.cs]: Oops! So sorry, I completely forgot about this! I’ll add it this afternoon :)

Comment (tintoy) [reply src/KubeClient/ResourceClients/DynamicResourceClient.cs]: (it’s 9am here, so in a couple of hours)

Comment (tintoy) [reply src/KubeClient/ResourceClients/DynamicResourceClient.cs]: Will get started on this in a couple of minutes :)

Comment (tintoy) [reply src/KubeClient/ResourceClients/DynamicResourceClient.cs]: How about this: 4c9dca0beb69bceb6bd18d75286ec818e6b33032

**Comment by tintoy** [comment]
Ok, are we good to merge?

**Comment by felixfbecker** [comment]
I think so. I am testing rn.

**Comment by tintoy** [comment]
No rush on this end - will create a new release when it’s ready (might be tomorrow morning - it’s 10pm here). Thanks for testing!

**Comment by felixfbecker** [comment]
Everything works, all tests passing! 🎉

**Comment by felixfbecker** [comment]
Would be great to merge so I can depend on the stable version and merge my PR :)

**Comment by tintoy** [comment]
What about the YAML converter - would you rather I get that in before merging / releasing, or after? I don’t imagine it will take long.

**Comment by felixfbecker** [comment]
Are you referring to #97? Those are separate issues to me, cause that only affects `ConvertTo-KubeYaml`. But I'm going to bed in a few minutes anyway :)

**Comment by tintoy** [comment]
Ok - merging and releasing :)

**Comment by tintoy** [comment]
Published v2.3.1 😄
